core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -1070,11 +1070,27 @@ class ControllerApis(
   }
 
   def handleAssignReplicasToDirs(request: RequestChannel.Request): CompletableFuture[Unit] = {
+    authHelper.authorizeClusterOperation(request, CLUSTER_ACTION)
     val assignReplicasToDirsRequest = request.body[AssignReplicasToDirsRequest]
-
-    // TODO KAFKA-15426
-    requestHelper.sendMaybeThrottle(request,
-      assignReplicasToDirsRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
-    CompletableFuture.completedFuture[Unit](())
+    val context = new ControllerRequestContext(request.context.header.data, request.context.principal,
+      OptionalLong.empty())
+    controller.assignReplicasToDirs(context, assignReplicasToDirsRequest.data).handle[Unit] { (reply, e) =>
```

Comment:
It's not necessary to use `handle` here. You can use `thenApply`. If it returns a future which was completed exceptionally, it will be handled in `ControllerApis.handle`

Reply:
Good point.

This change isn't really part of this PR. This PR builds on KAFKA-15361 (#14902) and KAFKA-15426 (#14863). This change is part of #14863, so this was addressed in [2874e11](https://github.com/apache/kafka/pull/14863/commits/2874e11f53c3a3fc73d480614f8d1cf695754bb8).